src/main/java/org/terasology/climateConditions/HypothermiaSystem.java

```diff
@@ -0,0 +1,110 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.climateConditions;
+
+import org.terasology.biomesAPI.OnBiomeChangedEvent;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.entity.lifecycleEvents.OnAddedComponent;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.prefab.PrefabManager;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.characters.CharacterMovementComponent;
+import org.terasology.logic.delay.DelayManager;
+import org.terasology.logic.delay.PeriodicActionTriggeredEvent;
+import org.terasology.logic.location.Location;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.logic.players.PlayerCharacterComponent;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.particles.components.generators.VelocityRangeGeneratorComponent;
+import org.terasology.physics.events.MovedEvent;
+import org.terasology.registry.In;
+
+@RegisterSystem(value = RegisterMode.AUTHORITY)
+public class HypothermiaSystem extends BaseComponentSystem {
+    public static final String VISIBLE_BREATH_ACTION_ID = "Visible Breath";
+
+    @In
+    private EntityManager entityManager;
+    @In
+    private DelayManager delayManager;
+    @In
+    private PrefabManager prefabManager;
+
+    private float thresholdHeight = 60f;
+    private int initialDelay = 5000;
+    private int breathInterval = 7000;
+
+    @ReceiveEvent(components = {PlayerCharacterComponent.class, CharacterMovementComponent.class})
+    public void observeDangerZone(MovedEvent event, EntityRef player, LocationComponent location, CharacterMovementComponent movement) {
+        //TODO: react on OnBiomeChangedEvent to handle the danger zone
+        float height = event.getPosition().getY();
+        float lastHeight = height - event.getDelta().getY();
+        if (height > thresholdHeight && lastHeight <= thresholdHeight) {
+            player.addOrSaveComponent(new HypothermiaComponent());
+        }
+        if (height < thresholdHeight && lastHeight >= thresholdHeight) {
+            if(player.hasComponent(HypothermiaComponent.class)){
+                player.removeComponent(HypothermiaComponent.class);
+            }
+        }
+    }
+
+    @ReceiveEvent
+    public void onHypothermia(OnAddedComponent event, EntityRef player, HypothermiaComponent hypothermia){
+        if(player.hasComponent(HypothermiaComponent.class)){
+            delayManager.addPeriodicAction(player, VISIBLE_BREATH_ACTION_ID, initialDelay, breathInterval);
+        }
```

Comment:
As in the other PR, having `HypothermiaComponent` ensures taht the entity has it.
```suggestion      
        delayManager.addPeriodicAction(player, VISIBLE_BREATH_ACTION_ID, initialDelay, breathInterval);
```